makeservices/makehttp

```diff
@@ -1,53 +1,58 @@
-#!/bin/bash
-
-if [ $# -ne 0 ]; then
-  echo "Usage: '$0'"
-  exit 1
-fi
-
-USER_SERVICES="/services/http/users/${USER:0:1}/$USER"
-
-create_symlink() {
-    ln -v -s "$USER_SERVICES" ~/public_html
-    chmod -v 755 "$USER_SERVICES"
-
-    if [ $? -eq 0 ]; then
-        echo
-        echo "public_html folder has been created successfully."
-    else
-        echo
-        echo "public_html folder was NOT created successfully."
-    fi
-}
-
-if [[ -e "$HOME/public_html" ]]; then
-    # Get the epoch time for renaming the old public_html link/folder
-    EXT=$(date "+%s")
-
-    if [[ -h "$HOME/public_html" ]]; then
-        SYMLINK_PATH=$(readlink $HOME/public_html)
-
-        if [[ "$SYMLINK_PATH" != "$USER_SERVICES" ]]; then
-            echo "Your symbolic link public_html is incorrect"
-            echo "Creating a new link and moving the old link to public_html.$EXT"
-            mv public_html public_html.$EXT
-
-            create_symlink
-            exit 2
-        else
-            echo "public_html already exists"
-            echo "Not creating symlink"
-            exit 3
-        fi
-    else
-        echo "A folder with data exists in $HOME/public_html"
-        echo "Moving the old folder to public_html.$EXT"
-        mv public_html public_html.$EXT
-
-        create_symlink
-        exit 4
-    fi
-else
-    create_symlink
-    exit 0
-fi
+#!/usr/bin/env python3
+"""Creates a symlink to the current user's web directory in their home folder.
+
+The symlink is called "public_html" by default and points to
+"/services/http/users/<first letter of username>/<username>". This script is
+idempotent: if the symlink exists and points to the correct path, makehttp
+takes no action; otherwise, it creates the symlink after removing broken
+symlinks or renaming any file or directory with the same path.
+"""
+import datetime
+import os
+import os.path
+import sys
+
+USER_SERVICES = '/services/http/users/{user[0]}/{user}'
+
+
+def usage(args):
+    return 'Usage: ' + args[0]
```

Comment:
Might be cool to print `__doc__` at the end so the useful docstring you wrote gets printed as well.